### PR TITLE
Manifest dir

### LINKED ISSUE #31: [releng] revisit what we build on xpi-manifest push or PR
Right now, we build every active xpi in the manifest on a xpi-manifest push or PR.

This was fine when we had low activity and one or two active xpis.
Now that we have quite a few more active xpis and higher activity, we're burning compute cycles that we may not need.

One solution may involve hashing up the xpi-manifest definition of each xpi, as well as the latest revision on the active branch, and only rebuilding if one or more of those change.

### PR DESCRIPTION
**Pull request by escapewindow**
This needs shipit changes to land at the same time.

Edit: I also landed cached-task support, which fixes #31 . Now we'll get a build+dep-signing task in PRs and on-push, but only if the manifest changes for a given build.

### CODE REVIEW

File: manifests/reset-search-default-extension.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+description: Extension asks users to reset their default search for some extensions
+repo-prefix: searchdefaultaddon
+active: true
+private-repo: true
+branch: main
+artifacts:
+  - web-ext-artifacts/reset-search-defaults.xpi
+addon-type: privileged
```
Comment (escapewindow) [inline manifests/reset-search-default-extension.yml:8]: I think this was `system`: https://github.com/mozilla-extensions/xpi-manifest/blob/master/xpi-manifest.yml#L240

**Review by escapewindow — COMMENTED** [review]
Thanks! Other than the two comments, looks good.
This will need to land at ~ the same time as the shipit patch rolls out.
I can't mark this as approved since I created the PR, but we can either get someone to rubberstamp it, or I can force merge it.

File: manifests/reset-search-default-extension.yml (hunk 1 above)
Comment (jmaher) [reply manifests/reset-search-default-extension.yml:8]: yes it was changed recently

**Comment by escapewindow** [comment]
Superseded by #97 .

**Pull request closed without merging**